lib/redacted_struct.rb

```diff
@@ -57,4 +51,74 @@ def pretty_print(q)
     }
   end
   # rubocop:enable all
+
+  alias_method :to_s, :inspect
+
+  def base_name_for_inspection
+    if is_a? Struct
+      "struct"
+    elsif is_a? Data
+      "data"
+    end
+  end
+
+  def base_type_for_pp
+    if is_a? Struct
+      Struct
+    elsif is_a? Data
```

Comment:
ditto about the reference
```suggestion
    else
```